File: graphql-test/src/main/java/com/junyss/graphqltest/people/model/dto/request/PeopleSearchRequestDto.java
package com.junyss.graphqltest.people.model.dto.request;

import com.junyss.graphqltest.common.enumtype.BloodType;
import com.junyss.graphqltest.common.enumtype.Role;
import com.junyss.graphqltest.common.enumtype.Sex;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PeopleSearchRequestDto {
	private Long peopleId;
	private Long teamId;
	private String lastName;
	private String firstName;
	private Sex sex;
	private BloodType bloodType;
	private Integer serveYears;
	private Role role;
	private String hometown;
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/people/model/dto/request/PeopleSearchRequestDto.kt
package com.junyharangstudy.kotlingraphqltest.api.people.model.dto.request

import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.BloodType
import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.Role
import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.Sex

class PeopleSearchRequestDto (
    val lastName: String?,
    val firstName: String?,
    val sex: Sex?,
    val bloodType: BloodType?,
    val serveYears: Int?,
    val role: Role?,
    val hometown: String?
)